### PR TITLE
Add reusable component for selecting a course.

### PR DESCRIPTION
**Pull request by richardpattinson**
## Context

Adding a new component for selecting courses as per the reusable pattern.

## Guidance to review

None

## Link to Trello card

https://trello.com/c/0b3x0EvO/3472-select-course-component
## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/provider_interface/select_course_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: form_object, url: form_path, method: :post do |f| %>
```
Comment (benswannack) [inline app/components/provider_interface/select_course_component.html.erb:3]: What is the plan for validating that at least one option is selected (and displaying an error summary)?

File: app/components/provider_interface/select_course_component.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module ProviderInterface
+  class SelectCourseComponent < ViewComponent::Base
+    include ViewHelper
+
+    attr_reader :form_object, :form_path, :page_title, :courses, :course_id
```
Comment (benswannack) [inline app/components/provider_interface/select_course_component.rb:5]: Is the course_id attr_reader used anywhere?

**Review by benswannack — COMMENTED** [review]
Looking good, a couple of questions inline

File: app/components/provider_interface/select_course_component.html.erb (hunk 1 above)
Comment (despo) [reply app/components/provider_interface/select_course_component.html.erb:3]: We plan to test that on the form object, but not until we start working on the next card.

File: app/components/provider_interface/select_course_component.rb (hunk 2 above)
Comment (despo) [reply app/components/provider_interface/select_course_component.rb:5]: It's used https://github.com/DFE-Digital/apply-for-teacher-training/pull/4241/files#diff-58c92cf3b3dacb6de8b03a18d15bef170341967100187863c5bbfdbd4944e793R4 to set and read the selected course.

Comment (benswannack) [reply app/components/provider_interface/select_course_component.rb:5]: Ah is that not done inside the form_object (with the attr_accessor)?

Comment (despo) [reply app/components/provider_interface/select_course_component.rb:5]: Ah you are right. We'll remove

**Review by cwrw — COMMENTED** [review]
I noticed that a pattern has evolved here, we would like to render a collection of radio buttons for an object of either locations/providers/courses etc. Could we have one component instead that accepts a collections, what are your thoughts?

**Comment by despo** [comment]
@cwrw good point. New PR introducing reusable collection select component and removing SelectProvider one:
https://github.com/DFE-Digital/apply-for-teacher-training/pull/4256

**Pull request closed without merging**